src/com/reader/AbstractReader.java

```diff
@@ -0,0 +1,43 @@
+package com.reader;
+
+import com.sudoku.SudokuBoard;
+
+import java.util.Scanner;
+
+/**
+ * Class that contains reader methods to read different types of Files
+ */
+public abstract class AbstractReader {
+    protected String filePath;
+    protected Scanner scanner = null;
+    public static final int ASCCI_ZERO = "0".charAt(0);
+
+    public AbstractReader(String filePath) {
```

Comment:
If this class is only an abstract reader for the sudoku inputs, you should name it accordingly the name that right now has it makes you think that this is a generic reader, but it is sudoku abstract reader